docs/randomblind.md

```diff
@@ -0,0 +1,189 @@
+Even though the credential contains five attributes, the issuer must only give
+four concrete values to construct the credential. For example, a requestor can
+start an issuance session to issue such a credential using the following
+command:
+
+```
+irma session --issue irma-demo.stemmen.stempas=test,test.com,14-12-2020,15-12-2020
+```
+
+Upon scanning the QR code, the IRMA app user is asked permission to perform the session and
+obtain this credential.  During the session, the issuer and user jointly decide on the
+value of the randomblind attributes, in this case the `votingnumber` attribute.
+We explain in detail how this comes to pass in the next section. The final
+value, which is at most 256 bits, ends up in the credential. The app displays
+this attribute in a [base62 representation](https://en.wikipedia.org/wiki/Base62).
+
+
+Before issuance               |  After issuance
+:----------------------------:|:----------------------------:
+![Permission](assets/rb_permission.jpg) | ![Card](assets/rb_card.jpg)
+
+## Cryptography
+In this section we will explain in detail how the issuer and the user are able
+to perform such an issuance session. Randomblindness of attributes only concerns
+the issuance protocol. The disclosure and signature protocols involving randomblind attributes
+are in every aspect identical to disclosing or signing using a normal attribute.
+
+We can assume that both parties have 
+a scheme in which the indices of the randomblind attributes within the
+credential type coincide. If this is not the case, either party cancels the
+session. In the following example, we only have one randomblind attribute, but
+the protocol can be extended straightforwardly to issue multiple randomblind
+attributes.
+
+The following requires a bit of background regarding Camenisch-Lysyanskaya (CL)
+signatures.  A short summary can be found
+[here](https://privacybydesign.foundation/pdf/Idemix_overview.pdf).  A more
+extensive description of the Idemix specification can be found
+[here](https://dominoweb.draco.res.ibm.com/reports/rz3730_revised.pdf).
+
+In the following, the variables $R, S, Z$ are public parameters defined by the
+Idemix public key of the issuer. The issuance goes as follows:
+- The user starts by sampling a random $v'$ and a random 255-bit integer $m_{r}'$ at
+  the index $r$ of the randomblind attribute.
+  This so-called *share* of the attribute remains secret, similar to
+  how the user's secret key remains secret during issuance. 
+  The user computes the commitment $U = S^{v'} R_0^{m_0} R_r^{m_{r}'} \mod n$.
+  Note that $m_0$ is always the user's secret key. This commitment is sent to
+  the issuer along with a [zero-knowledge proof](https://irma.app/docs/zkp/)
+  of $v', m_0$ and $m_{r}'$.
+
+- The issuer samples a random prime $e$. The issuer also samples $v''$ and $m_{r}''$ at random.
+  Since attribute values (the $m$'s) are at most 256 bits, $m_{r}''$ is also 255 bits (to not overflow).
```

Comment:
Do you think it it necessary to talk about sizes of `m_{r}'` and `m_{r}''` at all? I only included that to explain the overflowing part. It might be just an irrelevent technical detail.